src/kernel/arch/x86/arch.zig

```diff
@@ -251,6 +254,31 @@ pub fn haltNoInterrupts() noreturn {
     }
 }
 
+///
+/// Write a byte to serial port com1. Used by the serial initialiser
+///
+/// Arguments:
+///     IN byte: u8 - The byte to write
+///
+fn writeSerialCom1(byte: u8) void {
+    x86_serial.write(byte, x86_serial.Port.COM1);
+}
+
+///
+/// Initialise serial communication using port COM1 and construct a Serial instance
+///
+/// Return: serial.Serial
+///     The Serial instance constructed with the function used to write bytes
+///
+pub fn initSerial() Serial {
+    x86_serial.init(x86_serial.DEFAULT_BAUDRATE, x86_serial.Port.COM1) catch |e| {
+        panic(@errorReturnTrace(), "Failed to initialise serial: {}", .{e});
+    };
+    return Serial{
+        .write = writeSerialCom1,
+    };
```

Comment:
Maybe could move the runtime test from kmain to here to follow the convention in the init function.

Reply:
Doing the serial tests isn't arch-specific so I thought it would be best to keep it in common code. How about I write an init function in the main serial.zig and have that initialise and run the tests?

Reply:
Sure, just not a fan of calling the runtime tests from kmain, as it is part of the module not kmain. You get me :P

Reply:
Yeah you're right actually. It's nice to keep things encapsulated :)